### ISSUE TITLE
incorrect-package regex is not anchored and lets invalid package names pass

### ISSUE THREAD

**Issue by VasilevNStas** [objectionary/lints#1377]
## Problem

`incorrect-package` lets invalid package names pass because the second alternative of its regex is not anchored to the start of the string.

## Root cause

`src/main/resources/org/eolang/lints/metas/incorrect-package.xsl:17`:

```xml
<xsl:if test="not(matches($tail, $eo:package-regexp))">
```

with the pattern

```xml
<xsl:variable name="eo:package-regexp" select="concat('[a-z][a-z0-9]*(\.[a-z0-9]+)*|[\p{L}\p{M}\p{N}_]+(\.[...])*$', ...)"/>
```

The second alternative `[\p{L}\p{M}\p{N}_]+(\.[...])*$` has no leading `^`, and `matches()` in XPath is not anchored by default. Verified: `-bad` and `org..example` are accepted without a defect.

## Minimal example

```eo
+package -bad
+architect yegor@256.com
+version 0.0.1
+home https://github.com/objectionary
+spdx SPDX-FileCopyrightText: Copyright (c) 2026 Objectionary.com
+spdx SPDX-License-Identifier: MIT

[] > main
```

`+package -bad` does not trigger `incorrect-package`, although the name is clearly invalid.

## Expected behavior

The whole regex should be anchored (a single `^...$` expression, or `^` added to the second alternative), so that invalid package names are always caught.

## Related

- `metas/prohibited-package.xsl` — sibling rule for runtime-repo packages.

**Comment by 0crat** [comment]
@vasilevnstas Thanks for reporting a new bug! You've earned +12 points. By reporting bugs, you help our project improve its quality. If you find anything else in the repository that [doesn't look](https://www.yegor256.com/2018/02/06/where-to-find-more-bugs.html) as good as you might expect, [do not hesitate](https://www.yegor256.com/2014/04/13/bugs-are-welcome.html) to [report](https://www.yegor256.com/2018/04/24/right-way-to-report-bugs.html) it. Your running score is [+734](https://www.eolang.org/zerocracy/objectionary-vitals.html); don't forget to check your [Zerocracy account](https://www.zerocracy.com/) too).
